src/EnergyPlus/CTElectricGenerator.cc

```diff
@@ -316,8 +316,8 @@ namespace CTElectricGenerator {
 
             // Validate fuel type input
             state.dataCTElectricGenerator->CTGenerator(genNum).FuelType =
-                static_cast<Constant::eResource>(getEnumerationValue(Constant::eResourceNamesUC, AlphArray(11)));
-            if (state.dataCTElectricGenerator->CTGenerator(genNum).FuelType == Constant::eResource::Invalid) {
+                static_cast<Constant::eFuel>(getEnumerationValue(Constant::eFuelNamesUC, AlphArray(11)));
+            if (state.dataCTElectricGenerator->CTGenerator(genNum).FuelType == Constant::eFuel::Invalid) {
```

Comment:
Perfect! If these are required fields, then there is no need to do this `== Invalid` check because the epJSON parser already checks for this so you will never get here. Yes, the obvious next question is if it checks for this then why does it not just give us the enumeration ... we will get to that at some point I'm sure.

Reply:
I checked that it is not a required-field, I'll leave these lines.